[IMP] purchase: update manage vendor bills

brse-odoo opened the pull request:
This PR is targeting the manage vendor bills doc in Odoo's Purchase documentation (previously moved from Accounting section).

Performed complete rewrite to the document for 14.0 to be fwd-ported (targeted revision for 15.0 and 16.0 to follow).

- Completely rewrote the .rst file and added example flows
- Replaced and updated all screenshots
- Doc is more robust and comprehensive than before

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/4037).

brse-odoo commented:
Hi @jero-odoo and @larm-odoo! Whichever of you two is available to do so, this PR is ready for round one of peer review. Thank you in advance!

jero-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
+=============
 
-- Ordered quantities
+From the :guilabel:`Accounting` app, vendor bills can be processed and managed in batches. To do
```
```suggestion
Vendor bills can be processed and managed in batches in the :guilabel:`Accounting` app. To do
```

jero-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
-- or Received quantities
+From here, click the :guilabel:`Action` icon to export, delete, or send & print the bills; click
+the :guilabel:`Print` icon to print the invoices or bills; or click :guilabel:`Register Payment` to
+create and process payments for multiple vendor bills at once. Select the correct
```
```suggestion
create and process payments for multiple vendor bills at once. Select the appropriate
```

jero-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
-the bill will be linked to all the appropriate purchase orders.
+.. note::
+  Clicking :guilabel:`Register Payment` to create payments for vendor bills in batches will only
+  work for posted journal entries. So, all bills must have a :guilabel:`Posted` :guilabel:`Status`
```
```suggestion
  work for journal entries whose :guilabel:`Status` is set to :guilabel:`Posted`.
```

jero-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
+.. note::
+  Clicking :guilabel:`Register Payment` to create payments for vendor bills in batches will only
+  work for posted journal entries. So, all bills must have a :guilabel:`Posted` :guilabel:`Status`
+  to be processed in batches.
```
```suggestion
```

jero-odoo reviewed (commented):
@brse-odoo Hey Brandon, all set! There are a few places where it looks like the line break occurs a bit early, but it was a bit difficult to see on this view (especially since they usually would involve splitting guilabels or other markups). But I would take a second look at some of the shorter lines just in case. Other wise, let me know if you had any questions about any of the other comments. 👍

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
 
-The **Purchase** application allows you to manage your purchase orders,
-incoming products, and vendor bills all seamlessly in one place.
+A vendor bill is an invoice received for products and services that a company purchases from a
```
Good call!

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
+  The products and quantities in the purchase order are used to generate a draft bill.
+- :guilabel:`Received quantities`: a bill is created only *after* part of the total order has been
+  received. The products and quantities *received* are used to generate a draft bill. An error
+  message will appear if creation of a vendor bill is attempted without receiving anything.
```
Added a warning admonition below

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
+Activating :guilabel:`3-way matching` ensures that vendor bills are only paid once some (or all) of
+the products included in the purchase order have actually been received. To activate it, go to
+:menuselection:`Purchase --> Configuration --> Settings`, and scroll down to the
+:guilabel:`Invoicing` section. Then, click
```
Unfortunately no :-(

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,177 @@
-This button is outlined in red below:
+.. important::
+   :guilabel:`3-way matching` is *only* intended to work with the :guilabel:`Bill Control` policy
+   set to :guilabel:`Received quantities`.
```
Good call

brse-odoo commented:
@odoo/us-doc-review This PR is ready for content review

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,183 @@
 
-You can then change the default management method for the selected
-product to be based on either:
+Vendor bills can be processed and managed in batches in the :guilabel:`Accounting` app. To do this,
+go to :menuselection:`Accounting --> Vendors --> Bills`. Then, click the :guilabel:`checkbox` at
+the top left of the page under :guilabel:`Create`. This will select all existing vendor bills with
+a :guilabel:`Posted` or :guilabel:`Draft` :guilabel:`Status`.
```
Vendor bills can be processed and managed in batches in the :guilabel:`Accounting` app. 

To do this, go to :menuselection:`Accounting app --> Vendors --> Bills`. Then, click the :guilabel:`checkbox` at
the top left of the page, beside the :guilabel:`Number` column, under the :guilabel:`Create` button. This selects all existing vendor bills with a :guilabel:`Posted` (or :guilabel:`Draft`) :guilabel:`Status`.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,183 @@
+a :guilabel:`Posted` or :guilabel:`Draft` :guilabel:`Status`.
 
-- Ordered quantities
+From here, click the :guilabel:`Action` gear icon to export, delete, or send & print the bills;
+click the :guilabel:`Print` icon to print the invoices or bills; or click
+:guilabel:`Register Payment` to create and process payments for multiple vendor bills at once.
+Select the appropriate :guilabel:`Journal`, choose a :guilabel:`Payment Date` and
+:guilabel:`Payment Method`, and click :guilabel:`Create Payment`. This will create a list of
+journal entries all tied to their appropriate vendor bills.
```
Then, with all the existing bills selected, click the :guilabel:`Action` gear icon at the top of the page (in the middle, beneath the search bar) to reveal a number of different options. 

[insert image of that drop-down menu]


Those options are: :guilabel:`Export`, :guilabel:`Delete`, :guilabel:`Reverse`, :guilabel:`Resequence`, :guilabel:`Post entries`, :guilabel:`Send & print`, :guilabel:`Create Debit Note`, :guilabel:`Register Pament`, :guilabel:`Send Bills for digitalization`.

(*NOTE* might be smart to list all those options and briefly explain what each one means/represents. Not just a few.)

To create and process payments for multiple vendors at once, click :guilabel:`Register Payment` from the :guilabel:`Action` drop-down. When selected, a pop-up appears.

In this pop-up window, select the appropriate journal in the :guilabel:`Journal` field, choose a payment date in the :guilabel:`Payment Date` field, and choose a :guilabel:`Payment Method`. There is also the option to :guilabel:`Group Payments` on this pop-up, as well. 

When ready, click the :guilabel:`Create Payment` button, which creates a list of journal entries on a separate page. This list of journal entries are all tied to their appropriate vendor bills.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,183 @@
-Batch Billing
--------------
+.. note::
+   Clicking :guilabel:`Register Payment` to create payments for vendor bills in batches will only
+   work for journal entries whose :guilabel:`Status` is set to :guilabel:`Posted`.
```
The :guilabel:`Register Payment` option for vendor bills in batches will only work for journal entries 
   with a :guilabel:`Status` set to :guilabel:`Posted`.

ksc-odoo reviewed (commented):
Hey @brse-odoo just finished my peer review. Great work! If you have any questions, or need clarification on anything, don't hesitate to reach out. Thanks!

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,183 @@
 
-   #. Your vendor is in back-order and is sending you invoices as they ship the products.
-   #. Your vendor is sending you a partial bill or asking for a deposit.
+On the new :guilabel:`Vendor Bill`, click :guilabel:`Add Credit Note` or :guilabel:`Add Debit Note`
+to add credit or debit notes to the bill; add a :guilabel:`Bill Reference` number; then click
+:menuselection:`Register Payment --> Create Payment` to complete the :guilabel:`Vendor Bill`.
```
Good idea to add in more detail and clarification here. I provided a more in-depth rundown of pretty much every available option and field here, so that later in the doc when they pop up again, readers will be familiar.

brse-odoo commented:
@StraubCreative This PR has passed the second round of peer review by the us-doc-review team.

I implemented changes/fixes, squashed, and pushed up the revised version here. **This is ready for your final review whenever you have a chance. Thank you!**

samueljlieber reviewed (changes requested):
Hi @brse-odoo 👋 I have a number of technical changes on this PR as well as a few suggestions I'd like your input on. Please note I did not make suggestions through the entire doc since many items like early 100th char line breaks, :guilabel: & :menuselection: swaps, and :guilabel: use were repeated.

Let me know if you have any questions, otherwise Im looking forward to seeing you input and suggestions 🙂

Thank you!

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,253 @@
 
-1. You begin with a **Request for Quotation (RFQ)** to send out to your
-   vendor(s).
+To view and edit the default *Bill Control* policy and make changes to it, go to
```
this is italicized in this sense because we're not talking about the UI yet.
brse is hypothetically talking about the policy

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,253 @@
-Installing the Purchase and Inventory applications
---------------------------------------------------
+.. tip::
+   If a product needs a different control policy, the default *bill control* setting can be
```
this doesn't need any formatting actually
bill control policies/settings are not unique to Odoo, we're not on or giving instruction around a dashboard, and the term has already been introduced so italics are not needed here.

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,253 @@
-   If you create such a product, it is recommended to set the product type to **Service**.
+When products are received into a company's warehouse, receipts are created. Once the company
+processes the received quantities, they can choose to create a vendor bill directly from the
+wareouse receipt form. Depending on the *bill control* policy chosen in the settings, vendor bill
```
could go either way since we're referencing a heading in the settings.
up to brse

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,253 @@
 
-Purchasing products or services
--------------------------------
+To create and manage vendor bills on receipts using the *ordered quantities* bill control policy,
```
Don't think we use a gui here since we're not there on the screen. We're emphasizing the field/setting that we will arrive to after the dashboard instruction.

brse-odoo commented:
@StraubCreative I've made the requested technical changes as discussed and pushed up the revisions. This is ready for final review whenever you have a chance. Thank you!

samueljlieber commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,249 @@
-  :align: center
+Additionally, the bill :guilabel:`Amount`, :guilabel:`Payment Date`, and :guilabel:`Memo`
+(:dfn:`Reference Number`) can be changed from this pop-up. Once ready, click :guilabel:`Create
+Payment` to finish creating the :guilabel:`Vendor Bill`. Doing so causes a green **In Payment**
```
I think this banner could use a :guilabel: since it is an element that appears on the UI
```suggestion
Payment` to finish creating the :guilabel:`Vendor Bill`. Doing so causes a green :guilabel:`In Payment`
```

samueljlieber commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,249 @@
-Vendor Bill Matching
-====================
+On the warehouse receipt form, click :menuselection:`Validate --> Apply` to mark the
+:guilabel:`Done` quantities. Then, navigate back to the :guilabel:`Purchase Order` (via the
```
Arguable instance of whether or not to use a :guilabel: or :menuselection: here, I'd prefer a menuselection. @brse-odoo @StraubCreative  WDYT? 
```suggestion
:guilabel:`Done` quantities. Then, navigate back to the :menuselection:`Purchase Order` (via the
```

brse-odoo replied:
I went with the menuselection, I agree it makes sense here

samueljlieber commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,249 @@
 
-How product quantities are managed
-----------------------------------
+.. image:: manage/manage-auto-complete.png
+   :align: center
+   :alt: Auto-complete drop-down list on draft vendor bill.
```
may be good to indent this image so it is within the tip admonition

samueljlieber commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -2,226 +2,249 @@
-have not deleted the previous line items from the first purchase order,
-the bill will be linked to all the appropriate purchase orders.
+.. seealso::
+   :doc:`control_bills`
```
Please use a global relative path here rather than local relative

samueljlieber reviewed (changes requested):
Hi @brse-odoo I have just a few more technical changes and then we should be good to go, thank you for your continued hard work on this PR - we are almost there! Once these changes are made please tag me once more so I can move this to final review. Thank you in advance!

brse-odoo commented:
@samueljlieber I've implemented the technical changes you suggested and pushed up the revisions. Back to you -- thank you!

samueljlieber reviewed (approved):
Hi @brse-odoo great job on this PR - looks good to me 🙂 Passing it to final review!

samueljlieber commented:
Resolved merge conflict in https://github.com/odoo/documentation/commit/b702451352298aa3747c9a3762672a5c24903986 due to a custom anchor added on `manage.rst` in https://github.com/odoo/documentation/pull/3914

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/manage.rst:
```diff
@@ -4,226 +4,249 @@ Manage vendor bills
-  to keep track of their inventory levels. These options imply
-  stock management and will allow for receiving these kinds of
-  products.
+Create (and manage) vendor bills on receipts
```
Parenthesis are not needed here.
They're also not present on the other "create and manage" heading that appears later in the doc.
```suggestion
Create and manage vendor bills on receipts
```

StraubCreative reviewed (approved):
Looks good @brse-odoo 
One small thing on a heading for you to look at below.
Merge when you're ready 🔥 

Note for later and cc'ing @samueljlieber here:
I wonder if it's worth creating simple 101 "flow" docs for things like SOs, POs, etc. That way we don't have to repeat how to do things like create RFQs so often and can just link to a seealso? Can talk about it more...I just noticed a lot in this doc.

@robodoo delegate=brse-odoo

brse-odoo commented:
@robodoo r+
